test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/OpenIdConnectHandlerTests.cs

```diff
@@ -0,0 +1,766 @@
+        /// <param name="loggerFactory">custom loggerFactory</param>
+        /// <returns>The application builder</returns>
+        public static IApplicationBuilder UseCustomOpenIdConnectAuthentication(this IApplicationBuilder app, CustomConfigureOptions customConfigureOption, ILoggerFactory loggerFactory)
+        {
+            return app.UseMiddleware<CustomOpenIdConnectAuthenticationMiddleware>(customConfigureOption, loggerFactory);
+        }
+
+        /// <summary>
+        /// Adds the <see cref="OpenIdConnectAuthenticationMiddleware"/> into the ASP.NET runtime.
+        /// </summary>
+        /// <param name="app">The application builder</param>
+        /// <param name="options">Options which control the processing of the OpenIdConnect protocol and token validation.</param>
+        /// <param name="loggerFactory">custom loggerFactory</param>
+        /// <returns>The application builder</returns>
+        public static IApplicationBuilder UseCustomOpenIdConnectAuthentication(this IApplicationBuilder app, IOptions<OpenIdConnectAuthenticationOptions> options, ILoggerFactory loggerFactory)
+        {
+            return app.UseMiddleware<CustomOpenIdConnectAuthenticationMiddleware>(options, loggerFactory);
+        }
+    }
+
+    public class OpenIdConnectAuthenticationContext : IAuthenticateContext
+    {
+        public OpenIdConnectAuthenticationContext(string scheme = null)
+        {
+            AuthenticationScheme = scheme ?? OpenIdConnectAuthenticationDefaults.AuthenticationScheme;
+        }
+
+        public string AuthenticationScheme
+        {
+            get;
+            set;
+        }
+
+        public void Authenticated(ClaimsPrincipal principal, IDictionary<string, string> properties, IDictionary<string, object> description)
+        {
+        }
+
+        public void NotAuthenticated()
+        {
+        }
+    }
+
+    /// <summary>
+    /// Provides a Facade over IOptions
+    /// </summary>
+    public class Options : IOptions<OpenIdConnectAuthenticationOptions>
+    {
+        OpenIdConnectAuthenticationOptions _options;
+
+        public Options(Action<OpenIdConnectAuthenticationOptions> action)
+        {
+            _options = new OpenIdConnectAuthenticationOptions();
+            action(_options);
+        }
+
+        OpenIdConnectAuthenticationOptions IOptions<OpenIdConnectAuthenticationOptions>.Options
+        {
+            get
+            {
+                return _options;
+            }
+        }
+
+        /// <summary>
+        /// For now returns _options
+        /// </summary>
+        /// <param name="name">configuration to return</param>
+        /// <returns></returns>
+        public OpenIdConnectAuthenticationOptions GetNamedOptions(string name)
+        {
+            return _options;
+        }
+    }
+
+    public class CustomConfigureOptions : ConfigureOptions<OpenIdConnectAuthenticationOptions>
```

Comment:
Good Question. Extensibility.
I want to make sure that my class is called.
Currently, name isn't used, future tests will start making use of the name.